tests/api/core/strapi/document-service/publish.test.api.ts

```diff
@@ -24,6 +24,22 @@ describe('Document Service', () => {
   });
 
   describe('Publish', () => {
+    testInTransaction('Publishing keeps creator fields from the draft', async () => {
+      const articleDb = await findArticleDb({ title: 'Article1-Draft-EN' });
+
+      await publishArticle({ documentId: articleDb.documentId });
+
+      const publishedArticle = await findArticleDb({
```

Comment:
do you have the populateCreatorFields set to true in the ct & don't you need to add them into the populate of the find ?

Reply:
:O what ! didnt know that existed

Reply:
ah 🤦 it was not being populated, but the validation was expecting the value to not be null, and it was undefined.

should be fixed now :)